frontend/src/metabase/common/components/CollectionPicker/utils.ts

```diff
@@ -122,9 +119,9 @@ export const getPathLevelForItem = (
     return 0;
   }
 
-  const parentCollectionId = getParentCollectionId(
-    item?.effective_location ?? item?.location,
-  );
+  const parentCollectionId =
+    item.collection_id ??
+    getParentCollectionId(item?.effective_location ?? item?.location);
```

Comment:
do we have a collection util for doing this? i did something similar in the trash and now i'm wondering if i did it wrong given how you wrote this... might be preferable to make this a util function so people like you are doing the thinking for me 🧠

Reply:
we have `getNonRootParentId` - but it doesn't work with non collection api payloads 🙃 

I'm wary of trying to generalize this because collections data comes in so many shapes and sizes. we stuff partial collection payloads onto a lot of things and I'm afraid of tricking someone into thinking they can have confidence they have the right info to figure out the parent. in the collection picker, we can make a few assumptions and use it more safely.

one example: i have seen `isPersonalCollection` used incorrectly by people who think it can tell if something is in a personal collection, but it only works if you pass it a top-level personal collection.